src/core/picturesource.h

```diff
@@ -26,7 +26,7 @@ class PictureSource : public QObject
 {
     Q_OBJECT
   public:
-    explicit PictureSource( QObject *parent = nullptr );
+    explicit PictureSource( QObject *parent = nullptr, const QString &prefix = QString(), const QString &pictureFilePath = QString() );
```

Comment:
Can you add a docstring that explains the behavior with respect to pictureFilePath and emitting the signal? With explicit recommendations for subclasses, something like "subclasses which implement their own file dialog should provide an empty string for \a pictureFilePath and emit \a pictureReceived when appropriate."

Reply:
Done!